src/js/actions/transform.js

```diff
@@ -231,7 +235,42 @@ define(function (require, exports) {
         // Used to calculate the new top/left positions of all layers in relation to top/left
         // of the group
         var overallWidthRatio = size.w ? size.w / overallBounds.width : 0,
-            overallHeightRatio = size.h ? size.h / overallBounds.height : 0;
+            overallHeightRatio = size.h ? size.h / overallBounds.height : 0,
+            overallLeftOffset = 0,
+            overallTopOffset = 0,
+            direction;
+
+        // This is the case if the width input was changed
+        if (size.hasOwnProperty("w")) {
+            direction = refPoint.charAt(0);
+            switch (direction) {
+                case "l":
+                    break;
+                case "m":
+                    overallLeftOffset = (overallBounds.width - size.w) / 2;
+                    break;
+                case "r":
+                    overallLeftOffset = overallBounds.width - size.w;
+                    break;
+            }
+        } else if (size.hasOwnProperty("h")) {
+            // if the height input was changed
+            direction = refPoint.charAt(1);
+            switch (direction) {
+                case "t":
+                    break;
+                case "c":
+                    overallTopOffset = (overallBounds.height - size.h) / 2;
+                    break;
+                case "b":
+                    overallTopOffset = (overallBounds.height - size.h);
```

Comment:
nit: You are not using brackets for `case "r"`, so maybe remove brackets here to keep consistency?